watchman/views.py

```diff
@@ -55,109 +52,102 @@ def status(request):
                     if type(_check[_type]) == dict:
                         result = _check[_type]
                         if not result['ok']:
-                            http_code = settings.WATCHMAN_ERROR_CODE
+                            ok = False
                     elif type(_check[_type]) == list:
                         for entry in _check[_type]:
                             for result in entry:
                                 if not entry[result]['ok']:
-                                    http_code = settings.WATCHMAN_ERROR_CODE
-            response.update(_check)
+                                    ok = False
+            checks.update(_check)
 
-    if len(response) == 0:
-        raise Http404(_('No checks found'))
+    return checks, ok
 
-    return response, http_code, {WATCHMAN_VERSION_HEADER: __version__}
 
+@auth
+@json_view
+@non_atomic_requests
+def status(request):
+    checks, ok = run_checks(request)
 
-def ping(request):
-    _deprecation_warnings()
+    if not checks:
+        raise Http404(_('No checks found'))
 
-    return HttpResponse('pong', content_type='text/plain')
+    return checks, 200 if ok else settings.WATCHMAN_ERROR_CODE, {WATCHMAN_VERSION_HEADER: __version__}
```

Comment:
Would you mind pulling the `http_code` out into it's own variable directly above? Just makes this line a bit easier to parse I think (same for in `bare_status`).

Reply:
Fixed in 657f135

Reply:
@jamesmallen did this get pushed to this branch? I don't see the update on the PR.